jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java

```diff
@@ -436,10 +436,12 @@ public void matchCreated(MatchCreatedEvent event) {
                         kruntime.queueWorkingMemoryAction(new SignalManagerSignalAction(eventType, event));
                     } else if (ruleName.startsWith( "RuleFlowStateEventSubProcess-" ) 
                             || ruleName.startsWith( "RuleFlowStateEvent-" ) 
-                            || ruleName.startsWith( "RuleFlow-Milestone-" ) 
                             || ruleName.startsWith( "RuleFlow-AdHocComplete-" )
                             || ruleName.startsWith( "RuleFlow-AdHocActivate-" )) {
                         kruntime.queueWorkingMemoryAction(new SignalManagerSignalAction(ruleName, event));
+                    } else if (ruleName.startsWith("RuleFlow-Milestone-")) {
+                        String caseId = (String) kruntime.getEnvironment().get(EnvironmentName.CASE_ID);
```

Comment:
I'd add check if `kruntime.getEnvironment().get(EnvironmentName.CASE_ID)` is not null before appending it